pkg/api/graph/interfaces.go

```diff
@@ -0,0 +1,96 @@
+package graph
+
+import (
+	"github.com/gonum/graph"
+)
+
+// Marker is a struct that describes something interesting on a Node
+type Marker struct {
+	// Node is the optional node that this message is attached to
+	Node graph.Node
+	// RelatedNodes is an optional list of other nodes that are involved in this marker.
+	RelatedNodes []graph.Node
+
+	// Severity indicates how important this problem is.
+	Severity Severity
+	// Key is a short string to identify this message
+	Key string
+	// Message is a human-readable string that describes what is interesting
+	Message string
+}
+
+// Severity indicates how important this problem is.
+type Severity string
+
+const (
+	// InfoSeverity is interesting
+	InfoSeverity = "info"
+	// WarningSeverity is probably wrong, but we aren't certain
+	WarningSeverity = "warning"
+	// ErrorSeverity is definitely wrong, this won't work
+	ErrorSeverity = "error"
+)
+
+type Markers []Marker
+
+// MarkerScanner is a function for analyzing a graph and finding interesting things in it
+type MarkerScanner func(g Graph) []Marker
+
+func (m Markers) FilterBySeverity(severity Severity) []Marker {
+	ret := []Marker{}
+	for i := range m {
+		if m[i].Severity == severity {
+			ret = append(ret, m[i])
+		}
+	}
+
+	return ret
+}
+
+type SortedMarkerBySeverity []Marker
```

Comment:
`BySeverity` ? It's cleaner[1] and more idiomatic I guess (see first example in [sort](http://golang.org/pkg/sort/)). The same applies for `SortedMarkerByNodeID` and `SortedMarkerByKey`.

[1] 
```go
sort.Sort(BySeverity(markers))
```